service/discovery/azure/arm_template_test.go

```diff
@@ -0,0 +1,773 @@
+// Copyright 2021 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package azure
+
+import (
+	"fmt"
+	"io"
+	"k8s.io/apimachinery/pkg/util/json"
+	"net/http"
+	"testing"
+
+	"clouditor.io/clouditor/voc"
+	"github.com/jinzhu/copier"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockARMTemplateSender struct {
+	mockSender
+}
+
+func newMockArmTemplateSender() *mockARMTemplateSender {
```

Comment:
As done in `mockARMTemplateSender`, I would suggest to change all `Arm` occurrences to `ARM`. If a field or variable starts with `arm` (i.e. it is unexported), maybe try to avoid `arm` in the beginning, e.g. just leaving it out when the context makes it clear what it is.

Reply:
Done